cytoscape-matterjs.js

```diff
@@ -0,0 +1,428 @@
+;(function(){ 'use strict';
+
+  // registers the extension on a cytoscape lib ref
+  var register = function( cytoscape ){
+
+    if( !cytoscape ){ return; } // can't register if cytoscape unspecified
+
+    var defaults = {
+      // define the default options for your layout here
+      refreshInterval: 16, // in ms
+      refreshIterations: 10, // iterations until thread sends an update
+      fit: true,
+      preset: undefined,
+      gravity: -10,
+      globalAirFriction: 0.25,
+      clusters: [],
+      mass: [],
+      tickTimeout: 2000,
+      rescaleOn: 1,
+      updateOn: 1,
+      depthRestrict: true,
+    };
+
+    var extend = Object.assign || function( tgt ){
+      for( var i = 1; i < arguments.length; i++ ){
+        var obj = arguments[i];
+
+        for( var k in obj ){ tgt[k] = obj[k]; }
+      }
+
+      return tgt;
+    };
+
+    function Layout( options ){
+      this.options = extend( {}, defaults, options );
+    }
+
+    Layout.prototype.run = function () {
+      // cytoscape.js variables
+      var layout = this;
+      var options = this.options;
+      var cy = options.cy;
+      var eles = options.eles;
+      var nodes = eles.nodes();
+      var edges = eles.edges();
+
+      // matter.js aliases
+      var Engine = Matter.Engine;
+      var Render = Matter.Render;
+      var World = Matter.World;
+      var Body = Matter.Body;
+      var Bodies = Matter.Bodies;
+      var Bounds = Matter.Bounds;
+      var Events = Matter.Events;
+      var Runner = Matter.Runner;
+      var Query = Matter.Query;
+      var Composite = Matter.Composite;
+      var Constraint = Matter.Constraint;
+      var Common = Matter.Common;
+      var Vertices = Matter.Vertices;
+
+      // matter.js variables
+      var engine = Engine.create();
+      var render = Render.create({
+        element: document.body,
+        engine: engine,
+        width: 2400,
+        height: 2400
+      });
+      var world = engine.world;
+      var runner = Runner.create();
+
+      Matter.use('matter-gravity');
+      world.gravity.scale = 0;
+
+      // layout specific variables
+      var matterNodes = [];
+      var matterEdges = [];
+      var matterCenterSprings = [{ id: 'top', children: [] }];
+
+// +------------------------------------------------------------------------+ //
+// +-----------------ADDS NODES TO THE MATTER JS SIMULATION-----------------+ //
+// +------------------------------------------------------------------------+ //
+// +------------------------------------------------------------------------+ //
+
+      function addToCenterSprings(i) {
+        if (matterNodes[i].parent !== undefined) {
+          for (var j = 0; j < matterCenterSprings.length; j++) {
```

Comment:
It would be better to keep a ref of the node in the centre spring so you don't have to do N*N lookups